core/include/paging.h

```diff
@@ -76,7 +76,7 @@ static paddr_t get_pagebase(vaddr_t p, uint order)
 }
 
 #define INVALID_ADDR  ~(uint64_t)0
-#define INVALID_PFN   INVALID_ADDR
+#define INVALID_PFN   ((uint64_t)(~0ULL) >> 24) /* 40-bit field */
```

Comment:
nit: Any good reason to use `(uint64_t)` and `ULL` at the same time? If there is, please also update `INVALID_ADDR` above. Actually, `INVALID_PFN` can be derived from `INVALID_ADDR`:

```suggestion
#define INVALID_PFN   (INVALID_ADDR >> 24)  /* 40-bit field */
```

Reply:
For amd64 / i386 ABIs this ULL with uint64_t isn't required. I will follow the suggestion.